client/modules/IDE/pages/IDEView.jsx

```diff
@@ -77,265 +79,243 @@ function WarnIfUnsavedChanges() {
+    ide.justOpenedProject,
+    selectedFile.name,
+    selectedFile.content,
+    dispatch,
+    autosaveProject
+  ]);
 
-  render() {
-    return (
-      <RootPage>
-        <Helmet>
-          <title>{getTitle(this.props)}</title>
-        </Helmet>
-        <IDEKeyHandlers getContent={() => this.cmController.getContent()} />
-        <WarnIfUnsavedChanges currentLocation={this.props.location} />
-        <Toast />
-        <CmControllerContext.Provider value={{ current: this.cmController }}>
-          <Header
-            syncFileContent={this.syncFileContent}
-            key={this.props.project.id}
-          />
-        </CmControllerContext.Provider>
-        {this.props.ide.preferencesIsVisible && (
-          <Overlay
-            title={this.props.t('Preferences.Settings')}
-            ariaLabel={this.props.t('Preferences.Settings')}
-            closeOverlay={this.props.closePreferences}
-          >
-            <Preferences />
-          </Overlay>
-        )}
-        <main className="editor-preview-container">
-          <SplitPane
-            split="vertical"
-            size={this.state.sidebarSize}
-            onChange={(size) => this.setState({ sidebarSize: size })}
-            onDragFinished={this._handleSidebarPaneOnDragFinished}
-            allowResize={this.props.ide.sidebarIsExpanded}
-            minSize={125}
-          >
-            <Sidebar />
-            <SplitPane
-              split="vertical"
-              defaultSize="50%"
-              onChange={() => {
-                this.overlay.style.display = 'block';
-              }}
-              onDragFinished={() => {
-                this.overlay.style.display = 'none';
-              }}
-              resizerStyle={{
-                borderLeftWidth: '2px',
-                borderRightWidth: '2px',
-                width: '2px',
-                margin: '0px 0px'
-              }}
-            >
+  return (
+    <RootPage>
+      <Helmet>
+        <title>{getTitle(project)}</title>
+      </Helmet>
+      <IDEKeyHandlers getContent={() => cmRef.current?.getContent()} />
+      <WarnIfUnsavedChanges />
+      <Toast />
+      <CmControllerContext.Provider value={cmRef}>
+        <Header syncFileContent={syncFileContent} />
+      </CmControllerContext.Provider>
+      <MediaQuery minWidth={770}>
+        {(matches) =>
+          matches ? (
+            <main className="editor-preview-container">
               <SplitPane
-                split="horizontal"
-                primary="second"
-                size={
-                  this.props.ide.consoleIsExpanded ? this.state.consoleSize : 29
-                }
-                minSize={29}
-                onChange={(size) => this.setState({ consoleSize: size })}
-                allowResize={this.props.ide.consoleIsExpanded}
-                className="editor-preview-subpanel"
+                split="vertical"
+                size={ide.sidebarIsExpanded ? sidebarSize : 20}
```

Comment:
Since you have the ternary here, you can delete the `useEffect` on the sidebar size.  You can also simplify the initial state to `  const [sidebarSize, setSidebarSize] = useState(160);`.  The `sidebarSize` state just needs to store the size of the expanded panel.  Then you use either the current expanded `sidebarSize` or the collapsed size of `20`.

Same for the `consoleSize`.

Reply:
yeah